chaoslib/litmus/container_kill/containerd_chaos/crictl-chaos.yml

```diff
@@ -1,173 +1,159 @@
+  shell: >
+    kubectl get pod {{ chaos_pod_name }} -n {{ c_ns }} -o 
+    custom-columns=:.spec.serviceAccountName --no-headers
+  args:
+    executable: /bin/bash
+  register: chaos_service_account
+
+- name: "[Prepare]: Patch the chaoslib image"
+  template:
+    src:  /chaoslib/litmus/container_kill/containerd_chaos/containerd.j2
+    dest: /tmp/container-chaos-pod.yml
+  vars:
+    containerd_image: "{{ lib_image }}"
+    c_svc_acc: "{{ chaos_service_account.stdout }}"
     
-    # Setting pod_running_status to nil
-    - set_fact: 
-        pod_running_status: ""
+# Setting pod_running_status to nil
+- set_fact: 
+    pod_running_status: ""
         
-     # Containerd chaos pods creation is attempted for a total of 3 times, if it is not immediately schedulable due to transient node conditions
-     # If the containerd-chaos pod is not schedulable across these 3 tries, the experiment is failed with message indicating improper cluster state.
-    - include_tasks: /utils/common/create_chaos_pod.yml
-      vars:
-        pod_ns: "{{ c_ns }}"
-        c_path: "/tmp/containerd-chaos-pod.yml"
-        pod_label: "name=containerd-chaos-{{ run_id }}"
-      with_sequence: start=1 end=3
-
-    # Failing the execution, If containerd-chaos pod won't come to running state after three retries.
-    - fail:
-        msg: "crictl-chaos lib failed, Unable to create as containerd-chaos pod couldn't be scheduled on the {{ app_node }} node"
-      when: "pod_running_status is not defined or pod_running_status != 'Running'"
-
-    # chaosStartTimeStamp contains the start timestamp
-    - name: "[Prepare]: Getting the chaos start timestamp"
-      set_fact: chaosStartTimeStamp="{{lookup('pipe','date \"+%Y-%m-%d %H:%M:%S\"')}}"
-
-    - name: "[Wait]: Wait until the containerd-chaos pod is completed"
-      shell: >
-        kubectl get pods -l name=containerd-chaos-{{ run_id }} --no-headers -n {{ c_ns }}
-        --no-headers -o custom-columns=:status.phase
-      args: 
-        executable: /bin/bash
-      register: result
-      until: "result.stdout == 'Succeeded' or result.stdout == 'Failed'"
-      delay: 1
-      retries: "{{ c_duration|int + 10 }}"
-
-      # chaosCurrentTimeStamp contains the current timestamp
-    - name: "[Prepare]: Getting the chaos current timestamp"
-      set_fact: chaosCurrentTimeStamp="{{lookup('pipe','date \"+%Y-%m-%d %H:%M:%S\"')}}"
-
-    # chaosDiffTimeStamp contains the difference of current timestamp and start timestamp
-    # it will helpful to track the run duration of helper pod
-    - set_fact:
-        chaosDiffTimeStamp: "{{ ((chaosCurrentTimeStamp | to_datetime) - (chaosStartTimeStamp | to_datetime)).total_seconds()  }}"
-
-    - fail:
-        msg: "containerd-chaos pod failed"
-      when: "chaosDiffTimeStamp|int < c_duration|int"
+# Containerd chaos pods creation is attempted for a total of 3 times, if it is not immediately schedulable due to transient node conditions
```

Comment:
the comment here and elsewhere can be changed to reflect new name of the helper pod